Managers/EnemyManager.gd

```diff
@@ -53,6 +53,9 @@ func _ready() -> void:
 ## This calls the inner function that actually chooses the enemy group [br]
 ## This is done to save the packed scene in this EnemyManger before it is returned
 func choose_enemy_group() -> PackedScene:
+	if current_enemy_group_packed != null:
```

Comment:
technically once we check `current_enemy_group` and the other time `current_enemy_group_packed`
it's fine to do the test twice I think

Reply:
One should not exist without the other - in fact if it does things would break like they did before when _packed was not being saved. It's effectively check for the same thing and for the same reason. I don't really see reason to keep both